sql_it_test.go

```diff
@@ -704,6 +704,78 @@ func TestSQLService_ExecuteProvidedSuggestion(t *testing.T) {
 	})
 }
 
+func TestSQLService_ExecuteMismatchExpectedResultType(t *testing.T) {
+	it.SkipIf(t, "hz < 5.0")
+	it.SQLTester(t, func(t *testing.T, client *hz.Client, config *hz.Config, m *hz.Map, mapName string) {
+		ctx := context.Background()
+		sqlRead := fmt.Sprintf(`SELECT * FROM "%s"`, mapName)
+		sqlDelete := fmt.Sprintf(`DELETE FROM "%s" WHERE __key = 1`, mapName)
```

Comment:
I think this test is more complex than necessary. I would add an `sql` field to the struct for `testCases`. With that, a single iteration over test cases seems to be possible.

Reply:
refactored as suggested, https://github.com/hazelcast/hazelcast-go-client/pull/755/commits/05b40e579a3390e0330e27a49081bf0b662c1e19